src/Core/Scripting/OOJSStation.m

```diff
@@ -992,48 +999,58 @@ static JSBool StationAddShipToShipyard(JSContext *context, uintN argc, jsval *vp
 		return NO;
 	}
 	// ok, feel pretty safe to include this ship now
-	[shipyardDefinition setObject:shipInfo forKey:@"ship"];
+	[result setObject:shipKey forKey:SHIPYARD_KEY_SHIPDATA_KEY];
+	[result setObject:shipInfo forKey:SHIPYARD_KEY_SHIP];
 
 	// add an ID
 	Random_Seed ship_seed = [UNIVERSE marketSeed];
 	int superRand1 = ship_seed.a * 0x10000 + ship_seed.c * 0x100 + ship_seed.e;
 	uint32_t superRand2 = ship_seed.b * 0x10000 + ship_seed.d * 0x100 + ship_seed.f;
 	ranrot_srand(superRand2);
```

Comment:
I think this is a bug here. It ends up generating the same IDs for different ships. Possible solution:
```
- ranrot_srand(superRand2);
+ superRand2 &= Ranrot();
```